ipatests/test_integration/test_ipahealthcheck.py

```diff
@@ -1138,6 +1138,63 @@ def test_ds_configcheck_passwordstorage(self, modify_pwdstoragescheme):
                 assert "cn=config" in check["kw"]["items"]
                 assert error_msg in check["kw"]["msg"]
 
+    @pytest.fixture
+    def expire_cert_critical(self):
+        """
+        Fixture to expire the cert by moving the system date using
+        date -s command and revert it back
+        """
+        self.master.run_command(['date','-s', '+3Years'])
+        yield
+        self.master.run_command(['date','-s', '-3Years'])
+        self.master.run_command(['ipactl', 'restart'])
+
+    def test_nsscheck_cert_expired(self, expire_cert_critical):
+        """
+        This test checks that critical message is displayed
+        for NssCheck when Server-Cert has expired
+        """
+        msg = "The certificate (Server-Cert) has expired"
+        returncode, data = run_healthcheck(
+            self.master, "ipahealthcheck.ds.nss_ssl", "NssCheck",
+        )
+        assert returncode == 1
+        for check in data:
+            assert check["result"] == "CRITICAL"
+            assert check["kw"]["key"] == "DSCERTLE0002"
+            assert "Expired Certificate" in check["kw"]["items"]
+            assert check["kw"]["msg"] == msg
+
+    @pytest.fixture
+    def expire_cert_error(self):
+        """
+        Fixture to expire the cert by moving the system date using
+        date -s command and revert it back
+        """
+        self.master.run_command(['date','-s', '+2Years'])
+        yield
+        self.master.run_command(['date','-s', '-2Years'])
+        self.master.run_command(['ipactl', 'restart'])
+
+    def test_nsscheck_cert_expiring(self, expire_cert_error):
+        """
+        This test checks that error message is displayed
+        for NssCheck when Server-Cert is about to expire
+        """
+        msg = (
+            "The certificate (Server-Cert) will "
+            "expire in less than 30 days"
+        )
+        returncode, data = run_healthcheck(
```

Comment:
If certmonger renews this cert while this method is executing, the next test (`test_ipa_healthcheck_expiring`) is likely to fail or be disturbed.
I would rather do the following:
- remove the fixture `expire_cert_error`
- move `test_nsscheck_cert_expiring` inside the test `test_ipa_healthcheck_expiring` as an internal method (`def execute_nsscheck_cert_expiring():`)
- inside `test_ipahealthcheck_expiring`, call `execute_nsscheck_cert_expiring` right after the loop `for check in ("IPACertmongerExpirationCheck", "IPACertfileExpirationCheck",):`

This way, there is only one place where the date is moved just before expiration and all the related healthcheck checks are done inside the same place.

Reply:
@flo-renaud have removed the expire_cert_error fixture.
moved test_nsscheck_cert_expiring inside the test test_ipa_healthcheck_expiring as an internal method.